doc/whats_new/v1.6.rst

```diff
@@ -74,6 +74,14 @@ Changelog
     :pr:`123456` by :user:`Joe Bloggs <joeongithub>`.
     where 123455 is the *pull request* number, not the issue number.
 
+:mod:`sklearn.metrics`
+......................
+
+- |Fix| `matthews_corrcoef` behaviour in cases of constant predictions or targets was
+  modified to align with the limit value of the function and avoid returning 0 for
+  perfect predictions.
+  :pr:`28509` by :user:`Noam Keidar <redjest>`.
```

Comment:
Could you also acknowledge @marctorsoc here since this was already solved in #23183 but we only need to extract it.

Reply:
Gladly (: